### PR TITLE
Uniswap v2 adapter

### PR DESCRIPTION
**Pull request by bergarces**
## **Pre-merge author checklist**

- [ ] A brief description of the protocol and adapters is added to the PR
- [ ] Files outside the protocol folder are not being edited and, if they are, it's clearly explained why in the PR
- [ ] `update me` comments are removed
- [ ] Contracts used are verified for that chain block explorer
- [ ] Test cases with a block number have been added to the `testCases.ts` file for every relevant method that has been implemented
  - [ ] positions
  - [ ] prices
  - [ ] deposits
  - [ ] withdrawals
  - [ ] profits
  - [ ] tvl
  - [ ] apr
  - [ ] apy
- [ ] `getAddress` from `ethers`
  - [ ] Is used to parse hardcoded addresses
  - [ ] Is used to parse addresses that come from contract calls when it is not clear they'll be in checksum format
  - [ ] It is NOT used to parse addresses from input methods
  - [ ] It is NOT used to parse addresses from metadata
- [ ] For every adapter that extends `SimplePoolAdapter`
  - [ ] `getPositions` is not overwritten and, if it is, it's clearly explained why in the PR
  - [ ] `getProtocolTokenToUnderlyingTokenRate` is not overwritten and, if it is, it's clearly explained why in the PR
- [ ] If the adapters requires to fetch static data on-chain (e.g. pool ids, token metadata, etc)
  - [ ] The adapter implements `IMetadataBuilder`
  - [ ] The `buildMetadata` method is implemented with the `@CacheToFile` decorator
  - [ ] All the static data is stored within the metadata JSON file

### CODE REVIEW

File: src/adapters/uniswap-v2/products/pool/uniswapV2PoolAdapter.ts [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+        headers: { 'Content-Type': 'application/json' },
+        body: JSON.stringify({
+          operationName: 'tokenDayDatas',
+          variables: {
+            tokenAddr: '0x9ea3b5b4ec044b70375236a281986106457b20ef',
+            skip: 0,
+          },
+          query:
+            '{ pairs(first: 50 where: {reserveUSD_gt: "1000000", volumeUSD_gt: "50000"} orderBy: reserveUSD orderDirection: desc) {id token0 {id symbol name decimals} token1 {id symbol name decimals}}}',
+        }),
```
Comment (bergarces) [inline src/adapters/uniswap-v2/products/pool/uniswapV2PoolAdapter.ts]: First 50 pools by TVL with $50K of volume over 50K. Does that sound ok?

File: src/adapters/uniswap-v2/products/pool/uniswapV2PoolAdapter.ts [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+      [Chain.Ethereum]:
+        'https://api.thegraph.com/subgraphs/name/ianlapham/uniswap-v2-dev',
+    }
+
+    const response = await fetch(graphQueryUrl[this.chainId]!, {
+      method: 'POST',
+      headers: { 'Content-Type': 'application/json' },
+      body: JSON.stringify({
+        query: `{ pairs(first: ${numberOfPairs} where: {volumeUSD_gt: ${minVolumeUSD}} orderBy: reserveUSD orderDirection: desc) {id token0 {id} token1 {id}}}`,
+      }),
```
Comment (bergarces) [inline src/adapters/uniswap-v2/products/pool/uniswapV2PoolAdapter.ts:89]: First 50 pools by TVL with $50K of volume over 50K. Does that sound ok?

File: src/adapters/uniswap-v2/products/pool/metadata/ethereum.protocol-token.json [hunk 3]
```diff
@@ -0,0 +1,1002 @@
+{
+  "0x21b8065d10f73EE2e260e5B47D3344d3Ced7596E": {
+    "protocolToken": {
+      "address": "0x21b8065d10f73EE2e260e5B47D3344d3Ced7596E",
+      "name": "Uniswap V2",
+      "symbol": "UNI-V2",
```
Comment (bergarces) [inline src/adapters/uniswap-v2/products/pool/metadata/ethereum.protocol-token.json:6]: Do we want to artificially change the name and/or symbol of the protocol token? All of them are called the same.

Comment (bergarces) [reply src/adapters/uniswap-v2/products/pool/metadata/ethereum.protocol-token.json]: Done

**Review by jpsains — APPROVED** [review]

**Pull request merged**